### ISSUE TITLE
"None" value of `hovertext` is shown as String

### ISSUE THREAD

**Issue by pmercatoris** [plotly/plotly.py#1244]
In plotly 2.5.1, a None value of `hovertext` would result in the `hoverinfo` being skipped.
However, in 3.3.0, that is not the case. Instead, it will show the None as a string.

```python
import plotly.graph_objs as go

test = go.Scatter(
    x=[0,1],
    y=[0,1],
    hovertext=[None, 'test'],
    mode='lines',
    hoverinfo='x+text',
)
```
In 2.5.1, test is:
```
{'hoverinfo': 'x+text',
 'hovertext': [None, 'test'],
 'mode': 'lines',
 'type': 'scatter',
 'x': [0, 1],
 'y': [0, 1]}
```
In 3.3.0, test is:
```
Scatter({
    'hoverinfo': 'x+text', 'hovertext': ['None', 'test'], 'mode': 'lines', 'x': [0, 1], 'y': [0, 1]
})
```

Resulting in this: 

![screenshot_2018-10-25_12-53-18](https://user-images.githubusercontent.com/8655813/47496358-de1c8c00-d856-11e8-8954-a679bed3947d.png)

**Comment by jonmmease** [comment]
Thanks for the report @pmercatoris, I see what you mean.  It looks like we're inadvertently converting `None` values to strings during validation.

**Comment by jonmmease** [comment]
Done in #1249 and will be released in version 3.4.0. Thanks for taking the time to let us know!
